### DISCUSSION TITLE
Iframe elements do not have a title...  but this is coming from a adsense iframe...

### DISCUSSION THREAD

**Discussion by Elindo586** [GoogleChrome/lighthouse#14363] [Q&A]
I am getting a couple error below that I think are coming from the adsense ad I got...  

![adsense](https://user-images.githubusercontent.com/93617433/188908395-78d7b3da-6a52-4cd8-b077-0a205481a2cb.PNG)

^^^

The code I have for it is just a copy and paste from adsense... and I don't see where to add a title...

import { useEffect } from "react";

```
const AdBanner = () => {
  useEffect(() => {
    try {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    } catch (err) {
      console.log(err);
    }
  }, []);

  return (
    <>
      <ins
        className="adsbygoogle"
        style={{
          display: "block",
        }}
        data-ad-client="ca-pub-my number goes here"
        data-ad-slot="8603987287"
        data-ad-format="auto"
        data-full-width-responsive="true"
      ></ins>
      <script
        dangerouslySetInnerHTML={{
          __html: `adsbygoogle = window.adsbygoogle || []).push({});
          `,
        }}
      ></script>
    </>
  );
};

export default AdBanner;
```

Granted I also have a YouTube Iframe but that one has a title...

the url would be this one...
https://www.tu.biz/blog/english/pressure-compensated-pump-control-adjustment

pretty much every page with the adsense ad has the same error..

**Comment by adamraine** [comment]
Lighthouse is accurately reporting an iframe without a title.

In the future it may be possible to filter by 3P elements https://github.com/GoogleChrome/lighthouse/issues/13668, but ultimately there is still content on the page missing accessibility features. You are welcome to raise this as an issue to adsense.

**Reply by Elindo586** [reply]
would you happen to know what is the forum for adsense?

**Reply by adamraine** [reply]
You could try these:
https://support.google.com/adsense#topic=3373519
https://groups.google.com/g/adsense-api

**Comment by ennkh** [comment]
that happened to me too, I need to fix it https://www.computer-pdf.com/

**Comment by simeononsecurity** [comment]
Kind of a bandaid fix but I implemented the following to fix it

```html
<script defer>
  window.addEventListener("DOMContentLoaded", function() {
    try {
      var iframes = document.querySelectorAll(
        'iframe[src*="https://googleads.g.doubleclick.net"], iframe[data-google-container-id]'
      );

      iframes.forEach(function(iframe) {
        iframe.setAttribute("title", "google ads");
        iframe.setAttribute("role", "banner");

        if (iframe.hasAttribute("data-google-container-id") && iframe.complete) {
          setAttributes(iframe);
        } else {
          iframe.addEventListener("load", function() {
            setAttributes(iframe);
          });
        }
      });

      function setAttributes(iframe) {
        var containerId = iframe.getAttribute("data-google-container-id");
        iframe.setAttribute("title", "google ads");
        iframe.setAttribute("role", "banner");
      }
    } catch (error) {
      // Handle any errors that occur during the execution of the code
      console.error("An error occurred:", error);
    }
  });
</script>
```
